src/utils/EntryModel.hxx

```diff
@@ -41,389 +41,465 @@
     }
 
-    /** Copy the entry data into the middle of a buffer.
-     * @param buf pointer to destination buffer
-     * @param start_index starting index of buffer for the copy destination
-     * @param digits number of digits to copy
-     */
-    void copy_to_buffer(char *buf, int start_index, int digits)
+    /// Set the value to the maximum.
+    void set_max()
     {
-        HASSERT(digits <= digits_);
-        memcpy(buf + start_index, data_, digits);
+        set_value(valueMax_);
     }
 
-    /** Change the sign of the data.
-     */
+    /// Change the sign of the data.
     void change_sign()
     {
-        HASSERT(std::is_signed<T>::value);
-        if (hasInitial_)
+        if (value_ < 0)
         {
-            set_value(-get_value());
+            --size_;
         }
-        else if (data_[0] == '-')
+        value_ = -value_;
+        if (value_ < 0)
         {
-            memmove(data_, data_ + 1, digits_ - 1);
-            data_[digits_] = ' ';
+            ++size_;
         }
-        else
+    }
+
+    /// Clamp the value at the min or max.
+    /// @param force Normally, clamping doesn't occur if the entry is "empty".
+    ///              However, if force is set to true, we will clamp anyways.
+    virtual void clamp(bool force = false)
+    {
+        if (force || !empty_)
         {
-            memmove(data_ + 1, data_, digits_ - 1);
-            data_[0] = '-';
+            // puposely do not reset the numLeadingZeros_
+            empty_ = false;
+            if (value_ < valueMin_)
+            {
+                value_ = valueMin_;
+            }
+            else if (value_ > valueMax_)
+            {
+                value_ = valueMax_;
+            }
         }
-        clamp();
     }
 
-    /// Get the number of significant digits
-    /// @return number of significant digits
-    unsigned digits()
+    /// Pre-increment value. While this method does prevent wrap around of
+    /// the native type limits, it is incumbent on the caller to limit the
+    /// resulting number of digits.
+    T operator ++()
     {
-        return digits_;
+        if (value_ < std::numeric_limits<T>::max())
+        {
+            ++value_;
+        }
+        clamp();
+        return value_;
     }
 
-    /// Determine if this object is holding an initial or modified value.
-    /// @return true if holding an initial value, else false if modified
-    bool has_initial()
+    /// Pre-decrement value. While this method does prevent wrap around of
+    /// the native type limits, it is incumbent on the caller to limit the
+    /// resulting number of digits.
+    T operator --()
     {
-        return hasInitial_;
+        if (value_ > std::numeric_limits<T>::lowest())
+        {
+            --value_;
+        }
+        clamp();
+        return value_;
     }
 
-    /// Clamp the value at the min or max.
-    void clamp()
+protected:
+    /// Set the min and max boundaries supported by the EntryModel parameters.
```

Comment:
/// Set the min and max boundaries supported based on the maxSize_ (digit count).

Reply:
Fixed.